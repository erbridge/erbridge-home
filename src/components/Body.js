import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import Logo from './Logo';
import ScrollArea from './ScrollArea';

import './Body.css';

class Body extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    pointerPosition: PropTypes.shape({
      x: PropTypes.number.isRequired,
      y: PropTypes.number.isRequired,
    }),
  };

  render() {
    const { className, children, pointerPosition } = this.props;

    return (
      <div className={classnames(className, 'Body')}>
        <Logo
          className="Body__logo"
          focused
          pointerPosition={pointerPosition}
        />
        <ScrollArea
          className="Body__scrollarea"
          renderView={({ ...props }) => (
            <div className="Body__scrollarea__view" {...props} />
          )}
          renderTrackHorizontal={({ ...props }) => (
            <div
              className="Body__scrollarea__track Body__scrollarea__track--horizontal"
              {...props}
            />
          )}
          renderTrackVertical={({ ...props }) => (
            <div
              className="Body__scrollarea__track Body__scrollarea__track--vertical"
              {...props}
            />
          )}
          renderThumbHorizontal={({ ...props }) => (
            <div className="Body__scrollarea__thumb" {...props} />
          )}
          renderThumbVertical={({ ...props }) => (
            <div className="Body__scrollarea__thumb" {...props} />
          )}
          hideTracksWhenNotNeeded
        >
          <div className="Body__content">{children}</div>
        </ScrollArea>
      </div>
    );
  }
}

export default Body;
