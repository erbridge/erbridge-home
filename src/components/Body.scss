@import 'styles/colours';
@import 'styles/layout';
@import 'styles/transition-timings.scss';

@include wrapper-with-content(Body);

.Body {
  position: relative;
  overflow: hidden;
}

.Body__logo {
  position: absolute;
  left: 0;
  bottom: 0;
  max-width: 50%;
  max-height: 50%;
  transform: translate(-35%, 30%);
  opacity: 0.2;
}

.Body__scrollarea {
  width: 100%;

  // react-scrollarea overrides
  &:hover .scrollbar-container {
    opacity: 1;
  }

  .scrollbar-container {
    width: 12px !important;
    opacity: 0.5;
    transition: all 0.4s, opacity $hover-transition-duration,
      background $hover-transition-duration;

    .scrollbar {
      transition: background $hover-transition-duration;
    }

    &:hover {
      background: none !important;
      opacity: 1;

      .scrollbar {
        background: $red !important;
      }
    }

    .scrollbar {
      width: 100% !important;
      background: $grey !important;
    }
  }
}

.Body__scrollarea__content {
  display: flex;
  justify-content: center;
}

.Body__content {
  padding: 0 18px;
}
