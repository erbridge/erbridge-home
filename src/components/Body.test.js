import { shallow } from 'enzyme';
import React from 'react';

import Body from './Body';

it('renders correctly', () => {
  expect(
    shallow(
      <Body className="Test__body">
        <p>Test content</p>
      </Body>,
    ),
  ).toMatchSnapshot();
});
