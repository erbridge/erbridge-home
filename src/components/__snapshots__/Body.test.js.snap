// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Body"
>
  <Logo
    className="Body__logo"
    focused={true}
  />
  <ScrollArea
    className="Body__scrollarea"
    contentClassName="Body__scrollarea__content"
    smoothScrolling={true}
    verticalContainerClassName="scrollbar-container vertical"
  >
    <div
      className="Body__content"
    />
  </ScrollArea>
</div>
`;
